controllers/volumereplicationgroup_controller.go

```diff
@@ -1213,6 +1291,42 @@ func (v *VRGInstance) uploadPV(s3ProfileName string,
 	return nil
 }
 
+func (v *VRGInstance) deletePVFromS3Stores(pvc *corev1.PersistentVolumeClaim, log logr.Logger) error {
+	log.Info("Delete PV from s3 stores", "PV", pvc.Spec.VolumeName, "s3Profiles", v.instance.Spec.S3ProfileList)
+
+	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
+		if err := v.reconciler.PVDeleter.DeletePV(v, s3ProfileName, pvc); err != nil {
+			return fmt.Errorf("error deleting PV %s using profile %s, err %w", pvc.Name, s3ProfileName, err)
+		}
+	}
+
+	return nil
+}
+
+type ObjectStorePVDeleter struct{}
+
+func (ObjectStorePVDeleter) DeletePV(v interface{}, s3ProfileName string,
+	pvc *corev1.PersistentVolumeClaim) (err error) {
+	vrgName := v.(*VRGInstance).instance.Name
+	s3Bucket := constructBucketName(v.(*VRGInstance).instance.Namespace, vrgName)
+
+	v.(*VRGInstance).log.Info("Deleting PersistentVolume metadata from object store", "Profile Name", s3ProfileName)
+
+	objectStore, err := v.(*VRGInstance).reconciler.ObjStoreGetter.objectStore(
+		v.(*VRGInstance).ctx, v.(*VRGInstance).reconciler, s3ProfileName, vrgName)
+	if err != nil {
+		return fmt.Errorf("failed to get client for s3Profile %s, err %w",
+			s3ProfileName, err)
+	}
+
+	// Delete PV to object store
+	if err := objectStore.deleteObject(s3Bucket, pvc.Spec.VolumeName); err != nil {
```

Comment:
@ShyamsundarR found that the PV object was not deleted from the s3 store.  The reason for it is because the prefix is wrong.  The prefix starts with the `[object type + "/"`.  We didn't build the prefix that way.  A simple fix is this:
`keyPrefix := reflect.TypeOf(corev1.PersistentVolume{}).String() + "/" + pvc.Spec.VolumeName`
However, Shyam also asked the question "why don't we just delete the bucket?"  Definitely, we should.

Reply:
I will add a `deleteBucket()` to `ObjectStorer` interface.